src/cargo/ops/cargo_config.rs

```diff
@@ -4,20 +4,25 @@ use crate::util::config::{Config, ConfigKey, ConfigValue as CV, Definition};
 use crate::util::errors::CargoResult;
 use crate::{drop_eprintln, drop_println};
 use anyhow::{bail, format_err, Error};
+use clap::{ArgEnum, PossibleValue};
 use serde_json::json;
 use std::borrow::Cow;
 use std::fmt;
 use std::str::FromStr;
 
+#[derive(clap::ArgEnum, Copy, Clone)]
 pub enum ConfigFormat {
     Toml,
     Json,
     JsonValue,
 }
 
 impl ConfigFormat {
-    /// For clap.
-    pub const POSSIBLE_VALUES: &'static [&'static str] = &["toml", "json", "json-value"];
+    pub fn possible_values() -> impl Iterator<Item = PossibleValue<'static>> {
+        ConfigFormat::value_variants()
+            .iter()
+            .filter_map(ArgEnum::to_possible_value)
+    }
 }
 
 impl FromStr for ConfigFormat {
```

Comment:
If we move forward with this, we should switch the `FromStr`s to be implemented in terms of `ArgEnum` as well

For example, `clap_complete`s `Shell`:
```rust
impl FromStr for Shell {
    type Err = String;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        for variant in Self::value_variants() {
            if variant.to_possible_value().expect("No value is skipped").matches(s, false) {
                return Ok(*variant);
            }
        }
        Err(format!("Invalid variant: {}", s))
    }
}
```